### PR TITLE
Pivotal ID # 184246638: Backend docker image

### PR DESCRIPTION
**Pull request by msrada94**
https://www.pivotaltracker.com/story/show/184246638

Please include a list of the changes:
* add GitLab job to build and push submitter docker image

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -81,3 +83,28 @@ fire-caos-itest:
   script: gradle clean :submission:submission-webapp:itest -PenableFire=true --rerun-tasks --stacktrace
   variables:
     ITEST_FAIL_FACTOR: "8"
+
+build-fat-jar:
```
Comment (Juan-EBI) [inline .gitlab-ci.yml:87]: looking good lets make the following changes:

1. Lets create an specialized version docker.yml file where we put this pipelines.
2. Lets change this pipeline image name to be `submitter:$CI_COMMIT_SHA-SNAPSHOT`
3. Lets create an job exactly as build-push-docker-image but only executed on prod (check other pipelines on example) that publish image but without SNAPSHOT

**Review by Juan-EBI — COMMENTED** [review]
Looking great please check comment

File: .gitlab-ci.yml (hunk 1 above)
Comment (msrada94) [reply .gitlab-ci.yml]: changes done!

**Review by Juan-EBI — APPROVED** [review]
LGTM

**Pull request merged**